### ISSUE TITLE
New storage cleanup logic.

### ISSUE THREAD

**Issue by jMyles** [nucypher/nucypher#1554]
The `__del__` method of node storage classes can be (and is) called unexpectedly, with bad consequences.

If we create several (around 10) Characters without is_me in rapid succession, the ForgetfulNodeStorage object's `__del__` is called.

Prior to #1451, we weren't able to instantiate Characters fast enough to encounter this issue, but now it's a problem.

**Comment by michwill** [comment]
Wow, that sounds like a race condition we need to deal with properly. Also amazing that we are so fast that we can see it.

**Comment by derekpierre** [comment]
@jMyles is the "unexpectedness" of the `__del__` method being called, just an object scope issue? For your regression test, once an Enrico goes out of scope, its ForgetfulNodeStorage object is no longer needed and so is deleted, and therefore deletes its stored data since it is "forgetful".

For example, if I modify the test to instead be:
```python
    enricos = []
    for i in range(12):
        enricos.append(Enrico(policy_encrypting_key=enacted_federated_policy.public_key))

    assert IncorrectDeletionLogic.called_ever
```
the assertion fails since `IncorrectDeletionLogic.called_ever` is now `False`.

**Comment by jMyles** [comment]
Haha oh man, it's even less exciting than that: I was instantiating Characters as `is_me`.  Simply removing that fixes it.  Don't I feel sheepish.

However, there *is* still something spooky happening here,

So, I initially had the same impression as you: this *must* be happening at the *end* of the init of an anonymous object, right?  That's a reasonable time for this object to be garbage collected.

However, if you throw a breakpoint in `bad_deletion`, you'll see that that's not the case: it happens either during context switching of OpenSSL (during a call out to its bindings), or when we append the node_storage_function to the node class.

Either way, this is much less of an issue than I had supposed - it's really only a problem when more than one `Learner` is instantiated using the same directory for node and cert storage, and there's very rarely a time when it makes sense to do that.

I need to think a bit more about how to close this, but it's likely that we can just throw `is_me=False` in Finnegan's and think about the other stuff later.

Also: why the heck does `is_me` default to `True`?  [Who's hairbrained idea was that?!](https://github.com/nucypher/nucypher/commit/f704f93cbe67af8518bdeb22c0779dfb352e84fc#diff-bbee730b3a3659680b55419452bde052R31)

**Comment by derekpierre** [comment]
👍 

> Also, if you reduce the number created to around 5 or 6, you'll see that it reliably is never called
👀 👻

**Comment by KPrasch** [comment]
Takeaways here (do any of these need follow up Issues?): 

1. Ensure learners do not use the same directory for node and certificate storage (there's very rarely a time when it makes sense to do that)
2. Consider `Character.is_me` defaulting to `False`
3. Investigate unexpected/missing forgetful node storage deletion calls.

We still have this in the codebase:
```
    # TODO: Pending fix for 1554.
    # def __del__(self):
    #     shutil.rmtree(self._temp_certificates_dir, ignore_errors=True)
```
I propose simply removing this method since tmp files it will be cleaned up by the filesystem when default values are used here.

**Comment by derekpierre** [comment]
Closed via #3307 .
